src/main/java/org/openrewrite/kotlin/internal/KotlinTreeParserVisitor.java

```diff
@@ -4139,6 +4137,26 @@ private static PsiElement findLastChild(@Nullable PsiElement parent, Predicate<P
         return null;
     }
 
+    // Search for the last child that satisfies a given condition. If at any point, a child meets a break condition before the condition is satisfied, the function should return null.
+    @Nullable
+    private static PsiElement findLastChild(@Nullable PsiElement parent, Predicate<PsiElement> condition,  Predicate<PsiElement> breakCondition) {
```

Comment:
I suspect existing PSI utils can do this already -- were the existing utilities insufficient?

Reply:
good one, thanks, I will update

Reply:
Sounds good -- as we pay tech debt, I think it'll be helpful to replace most unnecessary methods with PSI utils.

Moving forward, it will be helpful/beneficial to consider what utilities already exist rather than adding new code. Generally, I consider what is available before adding code like findLastChild. It's helpful to do so, since it prevents unnecessary complexity, and uses code that is already tested. I'm fairly certain that findLastChild can be replaced, thanks in advance!

Reply:
> I didn't find one from `PsiTreeUtil` to do the equivalent thing below, @traceyyoshima , did you see any method to support this?

Ah, I can show you how to find utilities again if that helps -- I don't mean to look at `PsiTreeUtil`. As you know, there are many utilities in the Kotlin compiler, including for the PSI. Several forms of utils, utility classes, and classes with extension methods exist.